framework/auth/views.py

```diff
@@ -230,6 +255,92 @@ def confirm_email_get(token, auth=None, **kwargs):
     ))
 
 
+@collect_auth
+def confirm_user_get(auth=None, **kwargs):
+    """Called at login to see if there are emails to add or users to merge.
+    methods: GET
+    """
+    # user = User.load(kwargs['uid'])
+    user = auth.user
+    verified_emails = []
+    for token in user.email_verifications:
+        if user.confirm_token(token):
+            if user.email_verifications[token]['confirmed']:
+                try:
+                    user_merge = User.find_one(Q('emails', 'eq', user.email_verifications[token]['email'].lower()))
+                except NoResultsFound:
+                    user_merge = False
+
+                verified_emails.append({'address': user.email_verifications[token]['email'],
+                                        'token': token,
+                                        'confirmed': user.email_verifications[token]['confirmed'],
+                                        'user_merge': user_merge.email if user_merge else False})
+            else:
+                #todo migrate to remove this hack
+                user.email_verifications[token]['confirmed'] = False
+    user.save()
+    return verified_emails
+
+
+@collect_auth
+def confirm_email_remove(auth=None, **kwargs):
+    """Called at login if user cancels their merge or email add.
+    methods: PUT
+    """
+    user = auth.user
+    confirmed_email = request.json.get('address')
+    email_verifications = user.email_verifications.copy()
+    for token in user.email_verifications:
+        if user.email_verifications[token]['email'] == confirmed_email:
+            if user.confirm_token(token):
+                email_verifications.pop(token)
+    user.email_verifications = email_verifications
+    user.save()
+    return {
+        'status': 'success',
+        'removed_email': confirmed_email
+    }, 200
```

Comment:
It might be, I'm less familiar with our routing system than I'd like, but I think json responses are usually automatically wrapped with the appropriate status code, headers, etc.